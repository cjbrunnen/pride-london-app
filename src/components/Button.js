// @flow
import React from "react";
import { StyleSheet, TouchableOpacity, Linking } from "react-native";
import Text from "./Text";
import {
  eventDetailsBuyButtonColor,
  eventDetailsBuyButtonTextColor
} from "../constants/colors";

type Props = {
  text: string,
  url?: string
};

const onPress = url => {
  if (url && url.length > 0) {
    Linking.openURL(url);
  }
};

const Button = ({ text, url }: Props) => (
  <TouchableOpacity style={styles.button} onPress={() => onPress(url)}>
    <Text style={styles.text}>{text}</Text>
  </TouchableOpacity>
);
Button.defaultProps = {
  url: ""
};

const styles = StyleSheet.create({
  button: {
    height: 40,
    backgroundColor: eventDetailsBuyButtonColor,
    borderRadius: 4,
    alignItems: "center",
    justifyContent: "center"
  },
  text: {
    color: eventDetailsBuyButtonTextColor
  }
});

export default Button;
