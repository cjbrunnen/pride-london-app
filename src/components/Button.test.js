// @flow
import React from "react";
import { shallow } from "enzyme";
import Button from "./Button";

it("renders correctly", () => {
  const output = shallow(<Button text="a button" />);
  expect(output).toMatchSnapshot();
});
