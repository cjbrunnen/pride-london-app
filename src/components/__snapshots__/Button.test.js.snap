// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#D8D8D8",
      "borderRadius": 4,
      "height": 40,
      "justifyContent": "center",
    }
  }
>
  <Text
    markdown={false}
    style={
      Object {
        "color": "#FFFFFF",
      }
    }
    type="text"
  >
    a button
  </Text>
</TouchableOpacity>
`;
